pennylane/ops/qubit.py

```diff
@@ -1601,6 +1603,57 @@ def _matrix(cls, *params):
         return U
 
 
+class ControlledQubitUnitary(QubitUnitary):
+    r"""ControlledQubitUnitary(U, control_wires, wires)
+    Apply an arbitrary fixed unitary to ``wires`` with control from the ``control_wires``.
+
+    **Details:**
+
+    * Number of wires: Any (the operation can act on any number of wires)
+    * Number of parameters: 1
+    * Gradient recipe: None
+
+    Args:
+        U (array[complex]): square unitary matrix
+        control_wires (Sequence[int] or int): the control wire(s)
+        wires (Sequence[int] or int): the wire(s) the unitary acts on
+
+    **Example**
+
+    The following shows how a single-qubit unitary can be applied to wire ``2`` with control on
+    both wires ``0`` and ``1``:
+
+    >>> U = np.array([[ 0.94877869,  0.31594146], [-0.31594146,  0.94877869]])
+    >>> qml.ControlledQubitUnitary(U, control_wires=[0, 1], wires=2)
+    """
+    num_params = 1
+    num_wires = AnyWires
+    par_domain = "A"
+    grad_method = None
+
+    def __init__(self, *params, wires=None, do_queue=True, control_wires=None):
+        if control_wires is None:
+            raise ValueError("Must specify control wires")
+
+        wires = Wires(wires)
+        control_wires = Wires(control_wires)
+
+        if Wires.shared_wires([wires, control_wires]):
+            raise ValueError(
+                "The control wires must be different from the wires specified to apply the unitary"
+            )
+
+        wires = control_wires + wires
+
+        U = params[0]
+        padding = 2 ** len(wires) - len(U)
+        CU = block_diag(np.eye(padding), U)
```

Comment:
Should there perhaps be a check that `U` is of the correct size, i.e. that it can be applied on the supplied wires? Otherwise using a wrong sized unitary would _seemingly_ work without warnings, but wouldn't really make sense.

Reply:
That is a great idea! I've added: 4e22fbd9995ca1f667cf05856a229e1467a1d4a7